### DISCUSSION TITLE
rerender doesn't happen when status changes

### DISCUSSION THREAD

**Discussion by weidezhang** [TanStack/query#4996] [Q&A] [answered]
Hi, 

I'm wondering when the useQuery status change, why it doesn't trigger rerender ? I used the isSuccess property returned from useQuery but I don't see the useEffect gets called when data is successfully fetched.  Any help is appreciated ! 

<code>
const component = () => {
  const { data, isLoading, isSuccess } = useQuery(
    ['image'],
    async () => {
      //fetch some image 
    },
    onSuccess: ()=> {
        //some logic for success callback, this gets called with no problem.   
     } 
  );
React.useEffect(() => { 
   if( isSuccess) { console.log("success") } 
}
 return (
     if(isSuccess) {
       //render jsx for success case  
    } else
     {
       //render jsx for default cases 
     }
  )
}
</code>

**Comment by louis-young** [comment, marked as answer]
Hi 👋 

The query status changing _does_ trigger a re-render. The issue is that your effect has no dependency list, meaning it'll only run once.
